src/cmdc_tools/datasets/official/HHS/__init__.py

```diff
@@ -0,0 +1,93 @@
+import pandas as pd
+
+import us
+
+from ..base import ArcGIS, CountyData
+from ... import DatasetBaseNoDate
+
+
+class HHS(DatasetBaseNoDate, ArcGIS):
+    """
+    This scraper retrieves the hospital data provided by HHS at:
+
+      https://protect-public.hhs.gov/pages/hospital-capacity
+
+    It includes information on hospital capacity, hospital beds in use,
+    hospital beds in use by covid patients, icu capacity, icu beds in
+    use, and icu beds in use by covid patients
+    """
+
+    ARCGIS_ID = "qWZ7BaZXaP5isnfT"
+    table_name = "us_covid"
+    pk = '("vintage", "dt", "fips", "variable_id")'
+    has_fips = True
+    state_fips = 0  # Using 0 to denote that this is a national database
+    source = "https://protect-public.hhs.gov/pages/hospital-capacity"
+    provider = "HHS"
+
+    def _get_percent_reporting(self):
+        # TODO: See what pattern for the variable,
+        #       `fac_reporting_{}` will look like -- Will they add
+        #       new columns or will it always report the date from
+        #       yesterday?
+        crename = {
+            "state_fips": "fips",
+            "definitive_fac": "number_of_facilities",
+            "fac_reporting_20200720": "number_facilities_reporting",
+        }
+        # Read percent of facilities reporting data
+        df = (
+            self.get_all_sheet_to_df(
+                "Percentage_of_Facilities_Reporting_by_State", 0, 5
+            )
+            .rename(columns=crename)
+            .loc[:, crename.values()]
+        )
+
+        # Convert fips codes to integers
+        df["fips"] = df["fips"].astype(int)
+
+        out = df.melt(id_vars=["fips"], var_name="variable_name")
+
+        return out
+
+    def _get_hospital_census(self):
+        crename = {
+            "state_fips": "fips",
+            "total_inpatient_beds": "hospital_beds_capacity_count",
```

Comment:
Some of these variables don't seem to exist:

```
In [15]: foo = self.get_all_sheet_to_df("NHSN_Percentage_of_Inpatient_Beds_Occupied", 0, 5)                                                                                       

In [16]: list(foo)                                                                                                                                                                
Out[16]: 
['OBJECTID',
 'state_name',
 'state_fips',
 'state_abbr',
 'hhs_region',
 'inpatient_bed_utilization',
 'pct_inpatients_with_covid',
 'icu_bed_utilization',
 'icu_beds_used_estimate',
 'inp_beds_used_estimate',
 'inpatient_beds_used_covid_est',
 'Shape__Area',
 'Shape__Length']
```

Looks like we are missing: `total_inpatient_beds`, `total_icu_beds`

I think we could probably compute them as `total_icu_beds = icu_beds_used_estimate / icu_bed_utilization`

and `total_inpatient_beds = inp_beds_used_estimate / inpatient_bed_utilization`

Do we want to do that?

Reply:
They all originally existed :unamused: 

Let me do some digging tomorrow to see if they just moved them -- I'm working on dashboards that have hospital/testing data tonight.